File: vault-ios/VaultCommon/Features/Uploads/UploadFile.swift
import Foundation
import VaultMobile

public enum UploadFileData {
    case data(data: Data)
    case file(path: String, removeFileAfterUpload: Bool)
}

public struct UploadFile {
    let name: String
    let size: Int64?
    let data: UploadFileData
}

File: vault-android/app/src/main/java/net/koofr/vault/features/uploads/UploadFile.kt
package net.koofr.vault.features.uploads

import java.io.InputStream

sealed class UploadFileData {
    class Stream(val stream: InputStream) : UploadFileData()
    class Bytes(val bytes: ByteArray) : UploadFileData()
}

data class UploadFile(val name: String, val size: Long?, val data: UploadFileData)
